File: app/src/main/java/com/andyanika/translator/main/MainActivity.java
package com.andyanika.translator.main;

import android.os.Bundle;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import com.andyanika.translator.R;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;
import ru.terrakok.cicerone.NavigatorHolder;


public class MainActivity extends DaggerAppCompatActivity {
    @Inject
    MainActivityPresenter presenter;

    @Inject
    NavigatorHolder navigatorHolder;

    @Inject
    MainActivityNavigator navigator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(presenter);

        if (savedInstanceState == null) {
            presenter.navigateToTranslation();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        navigatorHolder.setNavigator(navigator);
    }

    @Override
    protected void onPause() {
        navigatorHolder.removeNavigator();
        super.onPause();
    }

    @Override
    public void onBackPressed() {
        navigator.back();
    }

    public int getFragmentContainerId() {
        return R.id.content_frame;
    }
}

File: app/src/main/kotlin/com/andyanika/translator/main/MainActivity.kt
package com.andyanika.translator.main

import android.os.Bundle
import com.andyanika.resources.ResourceImpl
import com.andyanika.translator.R
import core.interfaces.Resources
import com.github.terrakok.cicerone.NavigatorHolder
import com.google.android.material.bottomnavigation.BottomNavigationView
import org.koin.android.ext.android.inject
import org.koin.androidx.scope.ScopeActivity
import org.koin.androidx.scope.activityScope
import org.koin.core.parameter.parametersOf
import org.koin.core.scope.Scope

class MainActivity : ScopeActivity() {

    override val scope: Scope by activityScope()

    private val presenter: MainActivityPresenter by inject()
    private val navigatorHolder: NavigatorHolder by inject()
    private val navigator: MainActivityNavigator by inject { parametersOf(this) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.setOnNavigationItemSelectedListener(presenter)
        if (savedInstanceState == null) {
            presenter.navigateToTranslation()
        }

        scope.declare(ResourceImpl(this) as Resources)
    }

    override fun onResume() {
        super.onResume()
        navigatorHolder.setNavigator(navigator)
    }

    override fun onPause() {
        navigatorHolder.removeNavigator()
        super.onPause()
    }

    override fun onBackPressed() {
        navigator.back()
    }

    val fragmentContainerId: Int
        get() = R.id.content_frame
}
